base/ltkeys.dtx

```diff
@@ -576,20 +576,27 @@
 % \begin{macro}{\@@_options_loaded:n}
 % \begin{macro}{\@@_options_loaded:nn}
 % \changes{v1.0l}{2022/10/20}{Correct an argument for a message}
+% \changes{v1.0p}{2025-01-12}{Suppress load-only error for global options}
 %   Indicates that the load-time options for a package have been processed:
 %   once this has happened, make them unavailable either with a warning or
-%   an error.
+%   an error. We suppress the warning where the option is in the global list:
+%   there is a bit of gymnastics with the catcodes.
 %    \begin{macrocode}
 \cs_new_protected:Npn \@@_options_loaded:n #1
   {
+    \tl_set:Ne \l_@@_tmpb_tl { \tl_to_str:N \@classoptionslist }
```

Comment:
using classoptionlist here rather than one of the versions that's been filtered to drop the values (eg for unused option warnings) seems a bit suspect.  If I change the 1606 test file to have 

```
\documentclass[testoption=?]{article}
```

Then the warning comes back for duplicate loads

```

LaTeX Warning: Package "testpackage" has already been loaded: ignoring
(LaTeX)        load-time option "testoption".
```

which doesn't seem quite right?